File: app/src/main/java/tpcreative/co/qrscanner/common/BackupSingleton.java
package tpcreative.co.qrscanner.common;

public class BackupSingleton {
    private static BackupSingleton instance ;
    private BackupSingleton.BackupSingletonListener listener;
    public static BackupSingleton getInstance(){
        if (instance==null){
            synchronized (BackupSingleton.class){
                if (instance==null){
                    instance = new BackupSingleton();
                }
            }
        }
        return instance;
    }
    public void setListener(BackupSingleton.BackupSingletonListener listener){
        this.listener = listener;
    }
    public void reloadData(){
        if (listener!=null){
            listener.reloadData();
        }
    }
    public interface BackupSingletonListener{
        void reloadData();
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/common/BackupSingleton.kt
package tpcreative.co.qrscanner.common

class BackupSingleton {
    private var listener: BackupSingletonListener? = null
    fun setListener(listener: BackupSingletonListener?) {
        this.listener = listener
    }

    fun reloadData() {
        if (listener != null) {
            listener.reloadData()
        }
    }

    interface BackupSingletonListener {
        fun reloadData()
    }

    companion object {
        private var instance: BackupSingleton? = null
        fun getInstance(): BackupSingleton? {
            if (instance == null) {
                synchronized(BackupSingleton::class.java) {
                    if (instance == null) {
                        instance = BackupSingleton()
                    }
                }
            }
            return instance
        }
    }
}